instrumentation-api-incubator/src/main/java/io/opentelemetry/instrumentation/api/incubator/semconv/http/HttpClientExperimentalHttpParamsRedactionExtractor.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.incubator.semconv.http;
+
+import static io.opentelemetry.instrumentation.api.internal.AttributesExtractorUtil.internalSet;
+
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+import io.opentelemetry.instrumentation.api.semconv.http.HttpClientAttributesGetter;
+import io.opentelemetry.semconv.UrlAttributes;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+public class HttpClientExperimentalHttpParamsRedactionExtractor<REQUEST, RESPONSE>
+    implements AttributesExtractor<REQUEST, RESPONSE> {
+
+  private final HttpClientAttributesGetter<REQUEST, RESPONSE> attributesGetter;
+
+  private static final Set<String> PARAMS_TO_REDACT =
+      new HashSet<>(Arrays.asList("AWSAccessKeyId", "Signature", "sig", "X-Goog-Signature"));
+
+  public HttpClientExperimentalHttpParamsRedactionExtractor(
+      HttpClientAttributesGetter<REQUEST, RESPONSE> attributesGetter) {
+    this.attributesGetter = attributesGetter;
+  }
+
+  public static <REQUEST, RESPONSE>
+      HttpClientExperimentalHttpParamsRedactionExtractor<REQUEST, RESPONSE> create(
+          HttpClientAttributesGetter<REQUEST, RESPONSE> attributesGetter) {
+    return new HttpClientExperimentalHttpParamsRedactionExtractor<>(attributesGetter);
+  }
+
+  @Override
+  public void onStart(AttributesBuilder attributes, Context parentContext, REQUEST request) {
+    String urlFull = attributesGetter.getUrlFull(request);
+    String redactedUrlParameters = redactUrlParameters(urlFull);
```

Comment:
as far as I can tell you loose the scrubbing done in `HttpClientAttributesExtractor#stripSensitiveData`